sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -341,9 +341,20 @@ case class MaxOf(left: Expression, right: Expression) extends BinaryArithmetic {
   }
 
   override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
-    if (ctx.isNativeType(left.dataType)) {
+    if (ctx.isNativeType(dataType)) {
       val eval1 = left.gen(ctx)
       val eval2 = right.gen(ctx)
+      val result = if (dataType == BooleanType) {
+        s"${ev.primitive} = ${eval1.primitive} || ${eval2.primitive};"
```

Comment:
i think at first it needs to check whether eval1 or eval2 isNull. and then calculate primitive.